### DISCUSSION TITLE
How to convert a CKPT file into .nemo format?

### DISCUSSION THREAD

**Discussion by egorsmkv** [NVIDIA-NeMo/NeMo#7659] [Q&A] [answered]
I’ve got checkpoints files after fine tuning but would like to export them into .nemo format. How can I archive it?

**Comment by asierhv** [comment, marked as answer]
Recently got the same problem, here is the code i used to get a .nemo file from a .ckpt file:

checkpoint = torch.load(model_ckpt.ckpt) 
torch.save(checkpoint,"ckpt2.ckpt")
model = nemo_asr.models.EncDecCTCModel.load_from_checkpoint("ckpt2.ckpt")
model.load_state_dict(checkpoint['state_dict'], strict=True)
model.save_to(output_path)

if you have an issue with tokenizer you may need to indicate the path where you store it in the new checkpoint.cfg:
checkpoint["hyper_parameters"]["cfg"]["tokenizer"]["model_path"]=os.path.join(tokenizer_path,"tokenizer.model")
checkpoint["hyper_parameters"]["cfg"]["tokenizer"]["vocab_path"]=os.path.join(tokenizer_path,"vocab.txt")
checkpoint["hyper_parameters"]["cfg"]["tokenizer"]["spe_tokenizer_vocab"]=os.path.join(tokenizer_path,"tokenizer.vocab")

Good luck!

**Reply by titu1994** [reply]
This is great !

**Reply by egorsmkv** [reply]
Thank you!
